Assets/Scripts/Utils/DamageableCharacter.cs

```diff
@@ -0,0 +1,107 @@
+using Unity.VisualScripting;
+using UnityEngine;
+
+public class DamageableCharacter : MonoBehaviour, IDamageable
+{
+    public float _health = 3;
+    public bool _targetable = true;
+    public bool _invincible = false;
+
+    public bool isAlive = true;
+    public bool disableSimulation = false;
+    public bool canTurnInvincible = false;
+    public float invincibilityTime = 0.25f;
```

Comment:
I'm being annoying now :)). Please make these private and `[SerializeField]`. You already have the public properties, and the underlying values should remain private.

Reply:
what's the purpose of calling it [SerializedField]?
`Public fields are serializable by default, just like the areas you mark with SerializeField.`

Reply:
private setters. At first glance i dunno if i should be able to change them from outside of not